lib/modules/dosomething/dosomething_signup/dosomething_signup.module

```diff
@@ -579,25 +579,38 @@ function dosomething_signup_user_signup($nid, $account = NULL, $source = NULL, $
     $account = $user;
   }
 
-  // Insert signup.
-  if ($sid = dosomething_signup_create($nid, $account->uid, $source)) {
-    $node = node_load($nid);
+  if (variable_get('rogue_collection', FALSE)) {
+    $rogue_signup = dosomething_rogue_send_signup_to_rogue([
+      'campaign_id' => $nid,
+      'source' => $source,
+    ]);
+
+    // Make sure the signup exists before using it
+    if ($rogue_signup) {
+      $node = $node = node_load($nid);
+      dosomething_signup_set_signup_message($node->title);
+    }
+  } else {
+    // Insert signup.
+    if ($sid = dosomething_signup_create($nid, $account->uid, $source)) {
+      $node = node_load($nid);
 
-    // Opt into competition if auto opt-in is enabled for this campaign.
-    $auto_opt_in = dosomething_helpers_get_variable('node', $nid, 'enable_auto_competition_opt_in');
+      // Opt into competition if auto opt-in is enabled for this campaign.
+      $auto_opt_in = dosomething_helpers_get_variable('node', $nid, 'enable_auto_competition_opt_in');
```

Comment:
Should these lines be pulled out of the feature flag conditional (since competition & flash message should happen regardless)?